widget/gridwrap.go

```diff
@@ -0,0 +1,349 @@
+package widget
+
+import (
+	"fmt"
+	"math"
+	"sync"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/container"
+	"fyne.io/fyne/v2/data/binding"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+// Declare conformity with Widget interface.
+var _ fyne.Widget = (*GridWrap)(nil)
+
+// GridWrapItemID is the ID of an individual item in the GridWrap widget.
+type GridWrapItemID int
+
+// GridWrap is a widget with an API very similar to widget.List,
+// that lays out items in a scrollable wrapping grid similar to container.NewGridWrap.
+// It caches and reuses widgets for performance.
+type GridWrap struct {
+	widget.BaseWidget
+
+	Length     func() int                                      `json:"-"`
+	CreateItem func() fyne.CanvasObject                        `json:"-"`
+	UpdateItem func(id GridWrapItemID, item fyne.CanvasObject) `json:"-"`
+
+	scroller      *container.Scroll
+	itemMin       fyne.Size
+	offsetY       float32
+	offsetUpdated func(fyne.Position)
+}
+
+// NewGridWrap creates and returns a GridWrap widget for displaying items in
+// a wrapping grid layout with scrolling and caching for performance.
+func NewGridWrap(length func() int, createItem func() fyne.CanvasObject, updateItem func(GridWrapItemID, fyne.CanvasObject)) *GridWrap {
+	gwList := &GridWrap{BaseWidget: widget.BaseWidget{}, Length: length, CreateItem: createItem, UpdateItem: updateItem}
```

Comment:
There is no need to pass `widget.BaseWidget{}` here.
```suggestion
	gwList := &GridWrap{Length: length, CreateItem: createItem, UpdateItem: updateItem}
```